native/source/mixpanel/detail/worker.cpp

```diff
@@ -135,7 +135,13 @@ namespace mixpanel
             nanowww::Response response;
             if (client.send_request(request, &response))
             {
-                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "response: " + response.content());
+                // Lock on updating the network requests allowed time
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
```

Comment:
I'm not familiar with this but I guess since this mutex is in a block, it gets released when this block ends, right?